vault/audited_headers.go

```diff
@@ -120,8 +131,15 @@ func (c *Core) setupAuditedHeadersConfig() error {
 		}
 	}
 
+	// Ensure that we are able to case-sensitively access the headers;
```

Comment:
Yeah I was only meaning to deprecate this at some point since this loop may become more and more unnecessary with future releases.